ARCHITECTURE.md

```diff
@@ -20,19 +20,17 @@ maintaining this project.
 4. **Extensibility.** The parser is written to work with different use cases in
 our budding Rust GraphQL ecosystem, be it building schema-diagnostics for Rover,
 or writing out query planning and composition algorithms in Rust. These all have
-quite different requirements when it comes to CST manipulation. We wanted to
+quite different requirements when it comes to document manipulation. We wanted to
 make sure we account for them early on.
 
 ## `apollo-parser`
 
 `apollo-parser` is the parser crate of `apollo-rs`. Its job is to take GraphQL
-queries or schemas as input and produce an Concrete Syntax Tree (CST). Users of
-apollo-parser can then programmatically traverse the CST to get information
-about their input. 
+queries or schemas as string input and produce an Concrete Syntax Tree (CST).
+Users of apollo-parser can then programmatically traverse the CST to get information
+about their input.
 
-There are three main components of `apollo-parser`: the lexer, the parser, and the analyser. We already have the lexer and the parser, and the analyser is in the process of getting written.
-
-![An overview of apollo-parser diagram. We initially start of with input data. A Lexer performs lexical analysis on the data and produces tokens. Tokens get syntactically analysed by the parser, first into an untyped syntax tree, then into a typed syntax tree. "Future Work" indicates that a typed syntax tree will be semantically analysed by the "Analyser" to produce a semantic model.](images/apollo-parser-overview.png)
```

Comment:
This diagram showed "semantic analysis" as a "future work" piece of apollo-parser, but it now done in apollo-compiler. Perhaps it could be replaced with a new diagram, but for now I’ve removed it.